src/database.c

```diff
@@ -291,14 +331,23 @@ static void end_long_term_read_ops(void)
 static char *lt_read_db(const char *index, int only_check_key)
 {
 	int rc;
-	char *data;
+	char *data, *hash;
 	MDB_val key, value;
+    size_t len;
 
 	if (start_long_term_read_ops())
 		return NULL;
 
-	key.mv_data = (void *)index;
-	key.mv_size = strlen(index);
+    len = strlen(index);
+    if (len > MAX_PATH_LEN) { /* key size is greater than LMDB key limit */
+        hash = path_to_hash(index);
+        key.mv_data = (void *)(hash);
+        key.mv_size = gcry_md_get_algo_dlen(GCRY_MD_SHA256) * 2 + 1;
+    }
+    else {
+        key.mv_data = (void *)index;
+        key.mv_size = len;
+    }
```

Comment:
Generally, I'd just split this whole chunk into a common helper i.e. set_db_key(...) or whatever other name and call it both from `write_db()` and `lt_read_db()`.